packages/components/src/local/hex-grid/index.tsx

```diff
@@ -17,24 +17,17 @@ import multiPolyFromGeoJSON, { TerrainPolygons } from './helpers/multi-poly-from
 import { MapContext } from '../mapping'
 
 /* Import Types */
-import { SergeHex, SergeGrid, Route, NewTurnValues, PlanMobileAsset, SelectedAsset, RouteStore } from '@serge/custom-types'
+import { SergeHex, SergeGrid, Route, NewTurnValues } from '@serge/custom-types'
 import { LAYDOWN_TURN } from '@serge/config'
 
 /* Render component */
 export const HexGrid: React.FC<{}> = () => {
+  const props = useContext(MapContext).props
+  if (typeof props === 'undefined') return null
   const {
     gridCells, planningConstraints, setNewLeg, setHidePlanningForm,
     selectedAsset, viewAsRouteStore, viewport, polygonAreas
-  }: {
```

Comment:
You can define there any types what you want but real props may can not mutch with real (inputed data) types

https://github.com/serge-web/serge/pull/928/files#diff-42f0cdc7af2680f52905fe1b8f7f9429147aaa25c14346581c5544bc78c7269fL2
If you check this part you will se that we used any type, so if we will define props in right type (MappingContext) it will automatically works with already defined props, and will match inputed data, and when you will change MappingContext or imput data you will get right ts validation messages on erros

Reply:
Aah, brilliant. Thanks :-)